### ISSUE TITLE
Expected "some property" to come before "-styled-mixin0"

### ISSUE THREAD

**Issue by chytreg** [styled-components/stylelint-processor-styled-components#117]
I have opened an issue https://github.com/hudochenkov/stylelint-order/issues/41, but the author claims it's problem with interpolation. I have seen that you have introduced recently new comment syntax to handle edge cases.

Could you advise me how I could solve my issue?
Maybe we took a wrong approach in the project and should write the code in a different way.
Some code samples:

```js
import styled from "styled-components";

const Container = styled.div`
  ${props => {if (props.border) return "border-bottom: 1px solid #E3E6E7;";}} 
  display: flex;
  flex-wrap: wrap;
  justify-content: ${props => {
  switch (props.align) {
    case "center": return "center";
    case "right": return "flex-end";
    case "spaced": return "space-between";
    default: return "flex-start";
    }
  }};
  padding: 6px 0;
`;
```

```js
import styled from "styled-components";

import { mediaMin } from "utils/media_queries";

const Nav = styled.nav`
  border-right: 1px solid #C5D3D6;
  box-shadow: 2px 0 5px 0 rgba(158, 181, 183, 0.3);
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  padding-top: 10px;
  position: relative;
  transition: width 0.4s;
  width: 35px;
  z-index: 1;
  ${mediaMin.s`width: 150px;`}
`;
export { Nav };
```
I've tracked down that issue is casued by those dynamic extends:
```js
${props => {if (props.border) return "border-bottom: 1px solid #E3E6E7;";}} 
```
```js
${mediaMin.s`width: 150px;`}
```

**Comment by emilgoldsmith** [comment]
Hi @chytreg. The way our library works as we don't use the processing power to actually run all the javascript to figure out what the actual input from an interpolation would be we substitute relevant dummy values, and this won't change with our newly implemented (but not quite yet released, will be soon!) interpolation tagging, it will simply make our substitutions more accurate.

I see in the other issue that the problem here is that you're using an order stylelint rule and since our substitution in your case is an scss mixin, and I guess your rules say mixins have to be somewhere else in your code.

**Comment by emilgoldsmith** [comment]
Actually now that I think of it, when we publish you could use the `sc-custom` tag so you decide yourself what should be substituted as these kind of small edge cases are exactly what that tag is for.

So what you'd do is probably 
```js
${/* sc-custom 'border-bottom: 1px' */ if (props.border)...}
```

Which would stop your linting error. 

I don't imagine we would ever provide anything sleeker than that for your edge case I'm afraid, I'm inclined to close this issue now, but I'll just wait for a bit to see if there are any other replies from other members or you that would change that :)

**Comment by chytreg** [comment]
@emilgoldsmith thx for your feedback. It's fine for me to use new `sc-` syntax.
It looks like https://github.com/styled-components/stylelint-processor-styled-components/commit/0bc8560de023b5f7a9f2ba4eaf19faa40c9cf6db has been merged to master and has been released as 0.4.0. Unfortunately, it does not work for me. Am I doing something wrong?

```
stylelint-processor-styled-components@^0.4.0:
  resolved "https://registry.yarnpkg.com/stylelint-processor-styled-components/-/stylelint-processor-styled-components-0.4.0.tgz#6d9ebc25543d085f847ed32191ef994c6858fa54"
```
tried this:
```diff
 import styled from "styled-components";

 const Container = styled.div`
-  ${props => {if (props.border) return "border-bottom: 1px solid #E3E6E7;";}}
+  ${props => {/* sc-custom 'border-bottom: 1px' */ if (props.border) return "border-bottom: 1px solid #E3E6E7;";}}
   display: flex;
   flex-wrap: wrap
```

and 

```diff
 import styled from "styled-components";

 const Container = styled.div`
-  ${props => {if (props.border) return "border-bottom: 1px solid #E3E6E7;";}}
+  ${/* sc-custom 'border-bottom: 1px' */ props => {if (props.border) return "border-bottom: 1px solid #E3E6E7;";}}
   display: flex;
   flex-wrap: wrap
```

without success

**Comment by chytreg** [comment]
> It looks like 0bc8560 has been merged to master and has been released as 0.4.0.

I'm wrong, it has been merged but not release yet.

**Comment by chytreg** [comment]
OK, now I'm trying to get the code from master branch but smth does not work :/

```diff
@@ -159,7 +158,7 @@
     "stylelint": "^8.0.0",
     "stylelint-config-standard": "^17.0.0",
     "stylelint-order": "^0.6.0",
-    "stylelint-processor-styled-components": "^0.4.0",
+    "stylelint-processor-styled-components": "git+ssh://git@github.com/styled-components/stylelint-processor-styled-components#master",
```
next I do `yarn install`
but it fails on linting command:
```
 ./node_modules/.bin/stylelint app/frontend/react/containers/TopArea/styles.js
Error: Could not find "stylelint-processor-styled-components". Do you need a `configBasedir`?
```

**Comment by emilgoldsmith** [comment]
Yeah as I said in my edit above:
> EDIT: Regarding to publishing, the version I'm talking about is our quickly upcoming 1.0 version, we don't know exactly when it'll be out but I think it's very reasonable to expect it out next week.

I'm afraid I'm not of much help with grabbing npm packages from github, I never tried that before, sorry! You'll either have to google that or wait until we release which truly is getting really close, I think basically as soon as we merge #116.

I'll close this issue for now though, and hope you figure it out.
